pkg/clients/secrets/secrets.go

```
@@ -0,0 +1,138 @@
+}
+
+// NewService creates a new Service based on the *github.Client
+// returned by the NewClient SDK method.
+func NewService(token string) *Service {
+	c := ghclient.NewClient(token)
+	r := Service(c.Actions)
+	return &r
+}
+
+// CreateOrUpdateSec create or update repository secret in GitHub
+func CreateOrUpdateSec(ctx context.Context, cr *v1alpha1.SecretsParameters, name string, client client.Client, gh Service) (string, error) {
+	encryptedSecret, hash, err := callEncryptSecret(ctx, client, cr, name, gh)
+	if err != nil {
+		return "", err
+	}
+
+	if _, err := gh.CreateOrUpdateRepoSecret(ctx, cr.Owner, cr.Repository, encryptedSecret); err != nil {
+		return "", err
+	}
+
+	return hash, nil
+}
+
+// IsUpToDate check if encrypted value is up to date
+func IsUpToDate(ctx context.Context, client client.Client, p *v1alpha1.SecretsParameters, o *v1alpha1.SecretsObservation, name string, gh Service) (bool, error) {
+	var hash string
+	sec, _, err := gh.GetRepoSecret(ctx, p.Owner, p.Repository, name)
+	if err != nil {
+		return false, err
+	}
+
+	ref := xpv1.CommonCredentialSelectors{SecretRef: p.Value}
+	val, err := resource.ExtractSecret(ctx, client, ref)
+	if err != nil {
+		return false, err
+	}
+
+	if hash = generateHash(string(val)); len(val) == 0 {
+		hash = generateHash("test")
+	}
+
+	if hash != *o.EncryptValue {
+		return false, nil
+	}
+
+	if sec.UpdatedAt.String() != *o.LastUpdate {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// callEncryptedSecret setup encrypted secret and generates hash to store
+func callEncryptSecret(ctx context.Context, client client.Client, cr *v1alpha1.SecretsParameters, name string, gh Service) (*github.EncryptedSecret, string, error) {
+	publicKey, _, err := gh.GetRepoPublicKey(ctx, cr.Owner, cr.Repository)
+	if err != nil {
+		return nil, "", err
+	}
+
+	ref := xpv1.CommonCredentialSelectors{SecretRef: cr.Value}
+	val, _ := resource.ExtractSecret(ctx, client, ref)
```

I think it's never good to ignore errors